cmd/dump.go

```diff
@@ -194,8 +194,8 @@ func runDump(ctx *cli.Context) error {
 		log.Info("Skipping database")
 	} else {
 		tmpDir := ctx.String("tempdir")
-		if _, err := os.Stat(tmpDir); os.IsNotExist(err) {
-			fatal("Path does not exist: %s", tmpDir)
+		if err := os.MkdirAll(tmpDir, os.ModePerm); err != nil {
```

Comment:
It can't do so. An app shouldn't auto create all parent directories, it just make the app shit everywhere on user's filesystem.

There should be 2 kinds of tmp directories:

1. `dump` uses os.TempDir, without any change
2. `web` uses `AppDataPath/tmp` and we can auto create it since the AppDataPath is also managed by Gitea.

Reply:
![image](https://github.com/user-attachments/assets/d8550640-c269-46ef-8105-8209f6b09019)